### DISCUSSION TITLE
ActiveCampaign : Not getting any data from ActiveCampaign API

### DISCUSSION THREAD

**Discussion by jussipvainionpaa** [grafana/grafana-infinity-datasource#108] [Specific APIs] [answered]
Hi,

I'm having issues displaying data from ActiveCampaign API.

I have put the URL in the config and set custom "Api-Token" header where I input my api key.
When going to the dashboard I'm not getting any data or any errors shown.

I'm trying to retrieve list of contacts: https://developers.activecampaign.com/reference#list-all-contacts

The API is working correctly as I'm getting the list of contacts just fine in Postman but in Grafana-Infinity I'm just getting empty data.

I tested removing the Api-Token -> got 403 error so the token is working
I tested changing the URL endpoint -> got 404 not found error so the URL is working

What could I try next?

**Comment by yesoreyeram** [comment, marked as answer]
Here is the sample configuration. Let me know if this helps.

Settings:
* Leave the URL empty
* Add Just the header `Api-Token`
![image](https://user-images.githubusercontent.com/153843/126699858-47ec342d-b89a-449c-9082-6c4a72afa07c.png)

and then in the query. Hope this helps.

Note: Make sure to enter the root of objects array. In my case the API response have `{ contacts: [] }`. so I gave contacts.

![image](https://user-images.githubusercontent.com/153843/126700143-1e3e7431-2ff4-4f1c-b2d7-ae873a376d56.png)

**Comment by jussipvainionpaa** [comment]
Thank you so much! I was actually just missing the "root". Getting all the data I need now!
